src/DataItem.ts

```diff
@@ -211,33 +214,54 @@ export class DataItem implements BundleItem {
     }
     const item = new DataItem(buffer);
     const sigType = item.signatureType;
-    const tagsStart = item.getTagsStart();
-
+    const targetStart = item.getTargetStart();
+    const targetPresent = buffer[targetStart] === 1;
+    let anchorStart = targetStart + (targetPresent ? 33 : 1);
+    const anchorPresent = buffer[anchorStart] === 1;
+    const anchor = anchorPresent ? buffer.subarray(anchorStart + 1, anchorStart + 33) : Buffer.alloc(0);
+    if (anchor.length > 32) {
+      return false;
+    }
+    anchorStart += anchorPresent ? 33 : 1;
+    const tagsStart = anchorStart;
     const numberOfTags = byteArrayToLong(buffer.subarray(tagsStart, tagsStart + 8));
+    if (numberOfTags > MAX_TAGS) {
+      return false;
+    }
     const numberOfTagBytesArray = buffer.subarray(tagsStart + 8, tagsStart + 16);
     const numberOfTagBytes = byteArrayToLong(numberOfTagBytesArray);
-
-    if (numberOfTagBytes > MAX_TAG_BYTES) return false;
-
     if (numberOfTags > 0) {
       try {
         const tags: { name: string; value: string }[] = deserializeTags(
           Buffer.from(buffer.subarray(tagsStart + 16, tagsStart + 16 + numberOfTagBytes)),
         );
-
         if (tags.length !== numberOfTags) {
           return false;
         }
-      } catch (e) {
+        for (const t of tags) {
+          if (!t.name || !t.value) {
+            return false;
+          }
+          const nameLen = Buffer.byteLength(t.name);
+          const valueLen = Buffer.byteLength(t.value);
+          if (nameLen === 0 || nameLen > MAX_TAG_KEY_LENGTH) {
+            return false;
+          }
+          if (valueLen === 0 || valueLen > MAX_TAG_VALUE_LENGTH) {
+            return false;
+          }
```

Comment:
While this does respect the documented spec, there are existing apps and integrations using empty strings on name and value through Turbo

```suggestion
          if  nameLen > MAX_TAG_KEY_LENGTH) {
            return false;
          }
          if (valueLen > MAX_TAG_VALUE_LENGTH) {
            return false;
          }
```

Reply:
I appreciate the input.
Apparently there are more instances of other projects violating the spec, so I think the only real change that stands a chance of getting eventually merged is going to be one where we simply check that there are at most 128 tags and each tag (name+val) is 4KB at most.